torch/optim/optimizer.py

```diff
@@ -64,7 +64,7 @@ def _default_to_foreach(tensorlists: List[List[torch.Tensor]], differentiable: b
     all_tensors = []
     for tensorlist in tensorlists:
         all_tensors.extend(tensorlist)
-    return all(p.is_cuda for p in all_tensors)
+    return all(p is None or (p.is_cuda and type(p) == torch.Tensor) for p in all_tensors)
```

Comment:
nit-picking
```suggestion
    return all(p is None or (p.is_cuda and torch.is_tensor(p, torch.Tensor) for p in all_tensors)
```

Reply:
Hmmm torch.is_tensor is just an `isinstance`, right? I actually intend for it to be a strict type equals, because some subclasses (like Distributed Tensor)  do not have support foreach ops yet.